### ISSUE TITLE
TLS handshake remote error: tls: bad certificate

### ISSUE THREAD

**Issue by Montycarlo** [cruise-automation/k-rail#45]
```
Helm v3.0.2  
Kubernetes Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.10-gke.0", GitCommit:"a988db14950de3628f9e21773f3de0bf52485534", GitTreeState:"clean", BuildDate:"2019-12-11T22:37:55Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
K-rail 6cff60ea41d1acf9f4a87ab4a739a0eeaa1472c1 tag: v1.3.0
```

After a successful install (first by commenting out `webhooks.yaml`, installing so the pods can be provisioned, then uncommenting `webhooks.yaml` and upgrading), inspection of the logs shows a series of tls errors.

kubectl logs --namespace k-rail --selector name=k-rail
```
......
2020/01/20 07:56:36 http: TLS handshake error from 10.1.2.35:50702: remote error: tls: bad certificate
2020/01/20 07:56:36 http: TLS handshake error from 10.1.2.35:50704: remote error: tls: bad certificate
2020/01/20 07:56:55 http: TLS handshake error from 10.1.2.36:45366: remote error: tls: bad certificate
.....
```

Inspection of the k-rail-cert secret shows the certificate and private key are both well-formed.
```
kubectl get secret -n k-rail k-rail-cert -o yaml
```

I'm not quite sure where this is coming from.

**Comment by dustin-decker** [comment]
Thanks for the report. I added [this change](https://github.com/cruise-automation/k-rail/pull/46/files#diff-15fc89f40f7774694b6d334b8090e63cR83) to ensure that the Deployment redeploys when the cert secret changes. Since the CA is regenerated during each deploy, the secret and webhook got updated but the deployment may not have restarted. There is no need to comment out the webhooks config prior to deploying since the k-rail namespace is excluded from the webhook configuration.
